### PR TITLE
User Documentations & Add Search feature

### PR DESCRIPTION
**Pull request by gkushang**

### CODE REVIEW

File: documentation/package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
   "scripts": {
     "build": "gatsby build",
     "develop": "gatsby develop",
-    "deploy": "gatsby build && gh-pages -d public -b master",
+    "deploy": "gatsby build && gh-pages -d public -b master -r git@github.com:gkushang/gkushang.github.io.git",
```
Comment (limingli0707) [inline documentation/package.json:53]: Just curious, when you deploy to the github page, do you need to enter password? Wondering if anyone else is able to publish content to the site. If so, what if they mess up the website by publishing random content ;)

Comment (gkushang) [reply documentation/package.json:53]: I believe only Contributors can push and deploy.

File: documentation/src/pages/index.js [hunk 2]
```diff
@@ -144,31 +150,31 @@ const IndexPage = () => {
...
           title="Multibrowsers | Parallel" 
           description="Codeceptjs BDD provides an ability to run BDD Feaure files all in Parallel on Single browser or can run same set of tests in Parallel but on different browsers/OS combo (multile browers) on Sauce Labs. It offers greate Test Execution Coverage for your App."  
-          link="/execution/sauce-labs/on-multiple-browsers/" 
+          link="/5-execution/5-run-on-multi-browsers/" 
           image={multibrowsers}/>
         </Grid>
 
 
         <Grid item sm={3}>
         <AppCard
```
Comment (limingli0707) [inline documentation/src/pages/index.js]: I the indexPage component, there are repeatable <AppCard> components in the JSX render function. Thinking we can use loop to make the code simpler. 

In your case. You can do something like
 ```
const docMetaData=[obj1,obj2,obj3];
var AppCards = docMetaData.map(function(metaData){
return <AppCard title={metaData.title}></AppCard})
```

Then use AppCards in the return function {AppCards}

Hope the doc would be helpful https://thinkster.io/tutorials/iterating-and-rendering-loops-in-react

File: documentation/gatsby-config.js [hunk 3]
```diff
@@ -93,6 +93,20 @@ module.exports = {
         icon: `src/images/cucumber-logo.png`
       },
     },
-    `gatsby-plugin-offline`
+    `gatsby-plugin-offline`,
+    {
+      resolve: '@gatsby-contrib/gatsby-plugin-elasticlunr-search',
```
Comment (limingli0707) [inline documentation/gatsby-config.js:98]: Good job! One thought is to make the search more accurate is that in MD files, we can have keywords just like what you did for title and subtitle. Then in the resolvers here we can also search by keywords just add `keywords: node => node.frontmatter.keywords`

**Review by limingli0707 — APPROVED** [review]

**Review by limingli0707 — APPROVED** [review]

**Pull request merged**